beacon-chain/blockchain/process_block.go

```diff
@@ -379,6 +364,32 @@ func (s *Service) handleEpochBoundary(ctx context.Context, postState state.Beaco
 	return nil
 }
 
+func (s *Service) updateEpochBoundaryCaches(ctx context.Context, st state.BeaconState, blockRoot []byte) error {
+	copied, err := transition.ProcessSlotsUsingNextSlotCache(ctx, st, blockRoot, st.Slot()+1)
```

Comment:
this seems wrong, you will be processing the slots twice now, instead of slot + 1 it becomes slot + 2